### PR TITLE
File change monitoring on s3_auth (#8905)

### PR DESCRIPTION
**Pull request by moonchen**
1. Add a file change monitoring API to iocore.
2. Use it to add config change monitoring to s3_auth.  This can be used to rotate s3_auth secrets.

(Previously I made PR #8905 against 9.2.x by mistake.  It should have been against master branch.  This PR is rebased on top of master.  It also incorporates an autest change suggested by @bneradt.)

### CODE REVIEW

**Review by bneradt — APPROVED** [review]
Looks good. Thanks @moonchen .

**Comment by bneradt** [comment]
Looks like the gold file doesn't quite match curl's output:
https://ci.trafficserver.apache.org/job/Github_Builds/job/autest/1219/console

Probably good to replace lines that aren't strictly relevant to the test output with ``.

In case it's helpful, issues like this make me lean toward using Proxy Verifier in AuTests. It has a purpose built field verification mechanism rather than the generic gold comparison.

**Comment by moonchen** [comment]
My text editor trimmed trailing whitespace from the gold file.  It looks like the other gold file to match ATS log output also failed. I'm investigating.

**Comment by moonchen** [comment]
Autest failure should be fixed with latest push.  Thanks @bneradt

File: include/ts/apidefs.h.in [hunk 1]
```diff
@@ -554,6 +554,11 @@ typedef enum {
   TS_EVENT_SSL_CLIENT_HELLO  = 60207,
   TS_EVENT_SSL_SECRET        = 60208,
 
+  TS_EVENT_FILE_CREATED = 60300,
+  TS_EVENT_FILE_UPDATED = 60301,
+  TS_EVENT_FILE_DELETED = 60302,
+  TS_EVENT_FILE_IGNORED = 60303,
+
   TS_EVENT_MGMT_UPDATE = 60300
```
Comment (bneradt) [inline include/ts/apidefs.h.in:562]: Just checking: is it intentional that `TS_EVENT_FILE_CREATED` and `TS_EVENT_MGMT_UPDATE` have the same enum value?

**Review by bneradt — CHANGES_REQUESTED** [review]

File: include/ts/apidefs.h.in (hunk 1 above)
Comment (moonchen) [reply include/ts/apidefs.h.in]: Definitely not.  Thanks for catching that.  I've moved file events so they no longer conflict.

**Review by zwoop — CHANGES_REQUESTED** [review]
Couple of things to do before we can merge:

1. This needs to be discussed on the mailing list, as is the normal process for adding new plugin APIs. A short description of the new APIs, with the prototypes etc., and a justification.
2. These new APIs only work on Linux, which I feel reasonably strongly is not acceptable. I don't think we have any other APIs that are Linux only. So, we'll need to add kqueue support to this as well please :).

**Comment by zwoop** [comment]
Also, removing 9.2.x from this, we'll add this new API and plugin changes for 10.0.x.

**Comment by moonchen** [comment]
> Couple of things to do before we can merge:
> 
> 1. This needs to be discussed on the mailing list, as is the normal process for adding new plugin APIs. A short description of the new APIs, with the prototypes etc., and a justification.
> 2. These new APIs only work on Linux, which I feel reasonably strongly is not acceptable. I don't think we have any other APIs that are Linux only. So, we'll need to add kqueue support to this as well please :).

1. API description sent to mailing list.
2. kqueue implementation now added.

**Pull request closed without merging**